### PR TITLE
Support for lecun normal weight initialization

### LINKED ISSUE #2290: Add support for lecun normal weight initialization
### Motivation and description

Lecun normal initialization is needed (as far as I understand) to properly make self normalized neural networks.

Since Flux provides the selu activation function and alpha dropout, it would be nice to have lecun normal built in as well.

### Possible Implementation

Draws samples from a truncated normal distribution centered on 0 with stddev = sqrt(1 / fan_in) where fan_in is the number of input units in the weight tensor. (That is from tensorflow website)

### LINKED ISSUE #2491: Flux has no Lecun Normalization weight init function?
### Motivation and description

Flux has no buildin Lecun Normalization weight init function?

i can't found any information in doc, or shoud i implement my own version?

### Possible Implementation

_No response_

### PR DESCRIPTION
**Pull request by RohitRathore1**
feat: Add LeCun normal weight initialization

- Implemented `lecun_normal` function for weight initialization based on fan-in.
- Provided default initializer for flexibility.
- Marked `lecun_normal` as non-differentiable using ChainRulesCore.

It resolves issue #2290 

[Edit: closes #2290, closes #2491, closes #2299]

### PR Checklist

- [x] Tests are added
- [ ] Entry in NEWS.md
- [x] Documentation, if applicable

### CODE REVIEW

File: test/utils.jl [hunk 1]
```diff
@@ -192,6 +192,10 @@ end
     end
   end
 
+  @testset "lecun_normal" begin
+    m = lecun_normal(100, 100)
+  end
```
Comment (mcabbott) [inline test/utils.jl:197]: These are just taken from the doctests:
```suggestion
  @testset "lecun_normal" begin
    @test std(Flux.lecun_normal(10, 1000)) ≈ 0.032f0 rtol=0.01
    @test std(Flux.lecun_normal(1000, 10)) ≈ 0.317f0 rtol=0.01
  end
```
But it's also included in other tests above.

**Review by CarloLucibello — APPROVED** [review]
